From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Tue, 16 May 2023 10:04:03 +0200
Subject: [PATCH] Stop overwriting CMAKE_FIND_ROOT_PATH when cross compiling

PTXdist uses CMAKE_FIND_ROOT_PATH [1] to instruct CMake where to find
package information when cross-compiling. Stop overwriting this with
the contents of CMAKE_PREFIX_PATH [2] in the build system.

[1] https://cmake.org/cmake/help/v3.25/variable/CMAKE_FIND_ROOT_PATH.html
[2] https://cmake.org/cmake/help/v3.25/variable/CMAKE_PREFIX_PATH.html

This issue was introduced with commit e99e3c640ab7 ("cmake: Add
UPDATE_DEPS functionality"). It is similar to the same issue in
vulkan-validationlayers.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 scripts/CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/scripts/CMakeLists.txt b/scripts/CMakeLists.txt
index 5b979d43de59..3b1ddbdff685 100644
--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -113,8 +113,4 @@ if (MOLTENVK_REPO_ROOT)
     list(APPEND CMAKE_PREFIX_PATH ${MOLTENVK_REPO_ROOT})
 endif()
 
-if (CMAKE_CROSSCOMPILING)
-    set(CMAKE_FIND_ROOT_PATH ${CMAKE_PREFIX_PATH} PARENT_SCOPE)
-else()
-    set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH} PARENT_SCOPE)
-endif()
+set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH} PARENT_SCOPE)
